app/Http/Controllers/ProfileController.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\User;
+use App\Models\Movie;
+use App\Models\Review;
+use App\Models\MovieUser;
+use App\Http\Requests\ProfileRequest;
+
+class ProfileController extends Controller
+{
+    public function dashboard(User $user)
+    {
+        $user->loadCount('reviews', 'movies as watchlist_count');
+
+        $reviews = $user->reviews()
+            ->select('id', 'user_id','movie_id', 'title', 'rating', 'created_at')
+            ->with(['movie:id,title,poster'])
+            ->latest()
+            ->take(4)
+            ->get();
+
+        $watchlist = $user->movies()
+                          ->select('id', 'title', 'poster')
+                          ->take(3)
+                          ->get();
+
+        return view('profile/dashboard', [
+            'reviews' => $reviews,
+            'watchlist' => $watchlist,
+            'user' => $user
+        ]);
+    }
+
+    public function reviews(User $user)
+    {
+        $reviews = $user->reviews()
+            ->select('id', 'user_id','movie_id', 'title', 'rating', 'created_at')
+            ->with(['movie:id,title,poster'])
+            ->latest()
+            ->paginate(8);
+
+        return view('profile/reviews', [
+            'reviews' => $reviews,
+            'user' => $user
+        ]);
+    }
+
+    public function edit(User $user)
+    {
+        if(auth()->user()->id === $user->id || auth()->user()->is_admin) {
+            return view('profile/edit', [
+                'user' => $user
+            ]);
+        } else {
+            return redirect('/')
+                ->with('message', 'You don\'t have access to that page');
+        }
+    }
+
+    public function update(ProfileRequest $request, User $user)
+    {
+        if(auth()->user()->id !== $user->id) {
+            return redirect('/')
+                ->with('message', 'You don\'t have access to that function');
+        }
```

Comment:
This one, because it goes to the ProfileRequest, you can add this check to the authorize() method :D 

However, in your test, you should check for a 403 instead of a redirect, if you want to keep the redirect you can take a look at the `Requests/Review/Update.php` there is an example there.